### PR TITLE
Finalize UriHandler API

### PR DESCRIPTION
**Pull request by joaomoreno**
I've made the following changes as discussed in the API call:

- Improve documentation
- Rename to `UriHandler`
- `handleUri` now returns `ProviderResult<void>` and we log errors, if any

### CODE REVIEW

File: src/vs/vscode.d.ts [hunk 1]
```diff
@@ -6145,6 +6160,31 @@ declare module 'vscode' {
 		 * @returns a [TreeView](#TreeView).
 		 */
 		export function createTreeView<T>(viewId: string, options: { treeDataProvider: TreeDataProvider<T> }): TreeView<T>;
+
+		/**
+		 * Registers a [Uri handler](#UriHandler) capable of handling system-wide [Uris](#Uri).
+		 * In case there are multiple windows open, the topmost window will handle the Uri.
```
Comment (jrieken) [inline src/vs/vscode.d.ts]: use lower-case `uri` for consistency

File: src/vs/vscode.d.ts [hunk 2]
```diff
@@ -6145,6 +6160,31 @@ declare module 'vscode' {
...
+		 * A Uri handler is scoped to the extension it is contributed from; it will only
+		 * be able to handle Uris which are directed to the extension itself. The Uri format
+		 * is predetermined by the extension's identifier.
+		 *
+		 * For example, if the `vscode.git` extension registers a Uri handler, it will only
+		 * be allowed to handle Uris with the prefix `{scheme}://vscode.git`, in which `{scheme}`
+		 * is either `vscode` or `vscode-insiders`. All the following Uris are examples:
```
Comment (jrieken) [inline src/vs/vscode.d.ts]: This is tricky, we don't use product-names in the documentation. I usually use 'the editor' but something better is needed for those samples. 

Also, add some clarity or familiar uri-terms, like: The uri-scheme must be the product name, the uri-authority must be the extension id, the path, query, and fragment can be chosen freely.

**Comment by joaomoreno** [comment]
Pushed updates.

**Comment by joaomoreno** [comment]
Let's try to remove branding from the documentation.

**Comment by joaomoreno** [comment]
test

**Comment by joaomoreno** [comment]
another **comment**

**Comment by joaomoreno** [comment]
what

**Comment by joaomoreno** [comment]
@jrieken Removed product specific language from the docs. Up for final review.

File: src/vs/vscode.d.ts [hunk 3]
```diff
@@ -6203,13 +6218,36 @@ declare module 'vscode' {
...
+		 * @param handler The uri handler to register for this extension.
+		 */
+		export function registerUriHandler(handler: UriHandler): Disposable;
+
 		/**
 		 * Registers a webview panel serializer.
 		 *
-		 * Extensions that support reviving should have an `"onWebviewPanel:viewType"` activation event and
-		 * make sure that [registerWebviewPanelSerializer](#registerWebviewPanelSerializer) is called during activation.
+		 * Extensions that support reviving should have an`"onWebviewPanel:viewType"` activation event and
```
Comment (jrieken) [inline src/vs/vscode.d.ts]: bad change?

**Review by jrieken — COMMENTED** [review]
:+1:

**Comment by jrieken** [comment]
Seems to have an accidental change but apart from that 🚢 it.

**Comment by joaomoreno** [comment]
Good catch. I used the git merge ui for the first time. 🤦‍♂️

**Pull request merged**